exchanges/coinut/coinut_wrapper.go

```diff
@@ -430,68 +452,74 @@ func (c *COINUT) GetExchangeHistory(p currency.Pair, assetType asset.Item) ([]ex
+	if _, err = strconv.Atoi(o.ClientID); err != nil {
+		return submitOrderResponse, fmt.Errorf("%s - ClientID must be a number, received: %s", c.Name, o.ClientID)
+	}
+	err = o.Validate()
 	if err != nil {
 		return submitOrderResponse, err
 	}
 
-	clientIDUint := uint32(clientIDInt)
-
-	if !c.instrumentMap.IsLoaded() {
-		err = c.SeedInstruments()
+	if c.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
+		var response *WsStandardOrderResponse
+		response, err = c.wsSubmitOrder(&WsSubmitOrderParameters{
+			Currency: o.Pair,
+			Side:     o.OrderSide,
+			Amount:   o.Amount,
+			Price:    o.Price,
+		})
+		if err == nil {
+			submitOrderResponse.IsOrderPlaced = true
+		}
+		submitOrderResponse.OrderID = fmt.Sprintf("%v", response.OrderID)
+	} else {
+		var APIResponse interface{}
+		var clientIDInt uint64
+		isBuyOrder := o.OrderSide == order.Buy
+		clientIDInt, err = strconv.ParseUint(o.ClientID, 0, 32)
 		if err != nil {
 			return submitOrderResponse, err
 		}
-	}
-
-	currencyID := c.instrumentMap.LookupID(c.FormatExchangeCurrency(s.Pair,
-		asset.Spot).String())
-	if currencyID == 0 {
-		return submitOrderResponse, errLookupInstrumentID
-	}
+		clientIDUint := uint32(clientIDInt)
 
-	switch s.OrderType {
-	case order.Limit:
-		APIresponse, err = c.NewOrder(currencyID,
-			s.Amount,
-			s.Price,
-			isBuyOrder,
-			clientIDUint)
-	case order.Market:
-		APIresponse, err = c.NewOrder(currencyID,
-			s.Amount,
-			0,
-			isBuyOrder,
-			clientIDUint)
-	}
+		err = c.loadInstrumentsIfNotLoaded()
+		if err != nil {
+			return submitOrderResponse, err
+		}
 
-	switch apiResp := APIresponse.(type) {
-	case OrdersBase:
-		orderResult := apiResp
-		submitOrderResponse.OrderID = strconv.FormatInt(orderResult.OrderID, 10)
-	case OrderFilledResponse:
-		orderResult := apiResp
-		submitOrderResponse.OrderID = strconv.FormatInt(orderResult.Order.OrderID, 10)
-	case OrderRejectResponse:
-		orderResult := apiResp
-		submitOrderResponse.OrderID = strconv.FormatInt(orderResult.OrderID, 10)
-		err = fmt.Errorf("orderID: %d was rejected: %v",
-			orderResult.OrderID,
-			orderResult.Reasons)
-	}
+		currencyID := c.instrumentMap.LookupID(c.FormatExchangeCurrency(o.Pair,
+			asset.Spot).String())
+		if currencyID == 0 {
+			return submitOrderResponse, errLookupInstrumentID
+		}
 
-	if err == nil {
-		submitOrderResponse.IsOrderPlaced = true
+		APIResponse, err = c.NewOrder(currencyID, o.Amount, o.Price,
+			isBuyOrder, clientIDUint)
+		if err != nil {
+			return submitOrderResponse, err
+		}
+		responseMap := APIResponse.(map[string]interface{})
+		switch responseMap["reply"].(string) {
+		case "order_rejected":
+			return submitOrderResponse, fmt.Errorf("clientOrderID: %v was rejected: %v", o.ClientID, responseMap["reasons"])
+		case "order_filled":
+			orderID := responseMap["order_id"].(float64)
+			submitOrderResponse.OrderID = strconv.FormatFloat(orderID, 'f', -1, 64)
+			submitOrderResponse.IsOrderPlaced = true
```

Comment:
Would you be able to clarify in more detail what is wrong/what you would like changed?

Reply:
What I mean is that the total order was immediately match, but we only set IsOrderPlaced. Would be good to have another field to be set that says order fully matched = true so that we don't need to send another request to the endpoint to query the actual order to find out if its matched. Just speeds up the entire process if we get instantly filled.

Reply:
If that data isn't returned by other exchanges it comes with an implication that the order hasn't been filled. Its easy to add, even if outside of scope, but it depends on how it gets used

Reply:
Thanks for making that change!